language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/index/dao/BPackageSymbolDAO.java

```diff
@@ -0,0 +1,324 @@
+import org.ballerinalang.langserver.index.dto.BRecordTypeSymbolDTO;
+import org.ballerinalang.langserver.index.dto.OtherTypeSymbolDTO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * DAO for BPackageSymbol.
+ * 
+ * @since 0.983.0
+ */
+public class BPackageSymbolDAO extends AbstractDAO<BPackageSymbolDTO> {
+
+    public BPackageSymbolDAO(Connection connection) {
+        super(connection);
+    }
+
+    /**
+     * Insert a single entry to Index DB.
+     *
+     * @param dto DTO to insert in to the index DB
+     */
+    @Override
+    public int insert(BPackageSymbolDTO dto) throws LSIndexException {
+        String query = "INSERT INTO bLangPackage (name, orgName, version) VALUES (?, ?, ?)";
+        PreparedStatement statement = null;
+        ResultSet rs = null;
+
+        try {
+            statement = this.connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, dto.getName());
+            statement.setString(2, dto.getOrgName());
+            statement.setString(3, dto.getVersion());
+            statement.executeUpdate();
+            rs = statement.getGeneratedKeys();
+            List<Integer> keys = this.getGeneratedKeys(rs);
+            return keys.get(0);
+        } catch (SQLException e) {
+            throw new LSIndexException("Error while inserting BLang Package");
+        } finally {
+            this.releaseResources(rs, statement);
+        }
+    }
+
+    /**
+     * Insert a list of entries in Index DB.
+     *
+     * @param dtoList List of entries to be inserted
+     * @return {@link List}     List of generated IDs
+     * @throws LSIndexException Exception while index access
+     */
+    @Override
+    public List<Integer> insertBatch(List<BPackageSymbolDTO> dtoList) throws LSIndexException {
+        return null;
+    }
+
+    /**
+     * Get all the entries in the corresponding table.
+     *
+     * @return {@link List}     List of retrieved entries
+     * @throws LSIndexException Exception while index access
+     */
+    @Override
+    public List<BPackageSymbolDTO> getAll() throws LSIndexException {
+        List<BPackageSymbolDTO> pkgDTOs = new ArrayList<>();
+        String query = "SELECT id, name, orgName, version FROM bLangPackage";
+        PreparedStatement statement = null;
+        ResultSet rs = null;
+        try {
+            statement = this.connection.prepareStatement(query);
+            rs = statement.executeQuery();
+            while (rs.next()) {
+                BPackageSymbolDTO dto = new BPackageSymbolDTO.BPackageSymbolDTOBuilder()
+                        .setId(rs.getInt(1))
+                        .setName(rs.getString(2))
+                        .setOrgName(rs.getString(3))
+                        .setVersion(rs.getString(3))
```

Comment:
Both `version` and `orgName` are in the same column?

Reply:
addressed with 714e3dbb4ac6caa47c02da7acb09b3139462662a